src/hypercall.rs

```diff
@@ -84,13 +85,49 @@ pub fn unlink(mem: &MmapMemory, sysunlink: &mut UnlinkParams) {
 }
 
 /// Handles an open syscall by opening a file on the host.
-pub fn open(mem: &MmapMemory, sysopen: &mut OpenParams) {
-	unsafe {
-		sysopen.ret = libc::open(
-			mem.host_address(sysopen.name).unwrap() as *const i8,
-			sysopen.flags,
-			sysopen.mode,
-		);
+pub fn open(mem: &MmapMemory, sysopen: &mut OpenParams, file_map: &Option<UhyveFileMap>) {
+	// TODO: We could keep track of the file descriptors internally, in case the kernel doesn't close them.
+	let requested_path = mem.host_address(sysopen.name).unwrap() as *const i8;
+
+	// If the file_map doesn't exist, the provided path will be used instead.
+	// (i.e. host filesystem access).
+	if let Some(file_map) = file_map {
+		// Rust deals in UTF-8. C doesn't provide such a guarantee.
+		// In that case, converting a CStr to str will return a Utf8Error.
+		//
+		// See: https://nrc.github.io/big-book-ffi/reference/strings.html
+		let guest_path = unsafe { CStr::from_ptr(requested_path) }.to_str();
```

Comment:
I'd much prefer a checked safe variant here

Reply:
I don't see any better way of doing this in https://doc.rust-lang.org/std/ffi/struct.CStr.html

I presumed `to_str()` to be "good enough" given that the type is a `Result(&str, Utf8Error)` and basically relies on Rust itself to protect against malicious input, no?